### ISSUE TITLE
Expose modality to iOS accessibility

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#3821]
```
/*
 Informs whether the receiving view should be considered modal by accessibility. If YES, then 
 elements outside this view will be ignored. Only elements inside this view will be exposed.
 default == NO
 */
@property (nonatomic) BOOL accessibilityViewIsModal NS_AVAILABLE_IOS(5_0);
```

We should add support for this to support modal dialogs.

It's also possible that we could use this feature to fix #3819

**Comment by goderbauer** [comment]
We fixed this on the Dart-side by removing semantic nodes of elements "behind" a modal dialog from the semantic tree.
